compiler/rustc_hir_analysis/src/collect/type_of.rs

```diff
@@ -526,9 +526,13 @@ pub(super) fn type_of_opaque(
         Ok(ty::EarlyBinder::bind(match tcx.hir_node_by_def_id(def_id) {
             Node::Item(item) => match item.kind {
                 ItemKind::OpaqueTy(OpaqueTy {
-                    origin: hir::OpaqueTyOrigin::TyAlias { .. },
+                    origin: hir::OpaqueTyOrigin::TyAlias { in_assoc_ty: false },
                     ..
                 }) => opaque::find_opaque_ty_constraints_for_tait(tcx, def_id),
+                ItemKind::OpaqueTy(OpaqueTy {
+                    origin: hir::OpaqueTyOrigin::TyAlias { in_assoc_ty: true },
+                    ..
+                }) => opaque::find_opaque_ty_constraints_for_impl_trait_in_assoc_type(tcx, def_id),
```

Comment:
One question -- why does *this* need to be separate? Shouldn't this work with the regular `find_opaque_ty_constraints_for_tait`, as long as typeck uses the right def-ids collected from `ImplTraitInAssocTypeCollector`?

Reply:
Are we still checking the nested bodies of associated items? Pls check if this example (which constrains an ITIAT incorrectly inside of a closure body) still fails:

```rust
#![feature(impl_trait_in_assoc_type)]

trait Foo {
  type Assoc;
  fn bar() -> Self::Assoc;
}

impl Foo for () {
  type Assoc = impl Sized;
  fn bar() -> Self::Assoc {
    let closure = || -> Self::Assoc { let x: Self::Assoc = (); x };
    let _: i32 = closure();
    1i32
  }
}
```

Reply:
Doesn't need to be separate per se, but the new impl is so straight forward... and if the TAIT impl changes, we don't need to worry about the ITAIT impl at all.